src/Windows/Avalonia.Win32/WindowImpl.AppWndProc.cs

```diff
@@ -336,6 +372,176 @@ protected virtual unsafe IntPtr AppWndProc(IntPtr hWnd, uint msg, IntPtr wParam,
                             return IntPtr.Zero;
                         }
 
+                        break;
+                    }
+                case WindowsMessage.WM_POINTERDEVICECHANGE:
+                    {
+                        //notifies about changes in the settings of a monitor that has a digitizer attached to it.
+                        //https://docs.microsoft.com/en-us/previous-versions/windows/desktop/inputmsg/wm-pointerdevicechange
+                        break;
+                    }
+                case WindowsMessage.WM_POINTERDEVICEINRANGE:
+                case WindowsMessage.WM_POINTERDEVICEOUTOFRANGE:
+                    {
+                        //notifies about proximity of pointer device to the digitizer.
+                        //contains pointer id and proximity.
+                        //https://docs.microsoft.com/en-us/previous-versions/windows/desktop/inputmsg/wm-pointerdeviceinrange
+                        break;
+                    }
+                case WindowsMessage.WM_NCPOINTERUPDATE:
+                    {
+                        //NC stands for non-client area - window header and window border
+                        //As I found above in an old message handling - we dont need to handle NC pointer move/updates.
+                        //All we need is pointer down and up. So this is skipped for now.
+                        break;
+                    }
+                case WindowsMessage.WM_NCPOINTERDOWN:
+                case WindowsMessage.WM_NCPOINTERUP:
+                case WindowsMessage.WM_POINTERDOWN:
+                case WindowsMessage.WM_POINTERUP:
+                case WindowsMessage.WM_POINTERUPDATE:
+                    {
+                        GetDevicePointerInfo(wParam, out var device, out var info, ref timestamp);
+
+                        if (info.historyCount > 1)
+                        {
+                            if (info.pointerType == PointerInputType.PT_TOUCH)
+                            {
+                                if (ShouldIgnoreTouchEmulatedMessage())
```

Comment:
Why aren't we using WM_POINTER messages for touch input too?

Reply:
BTW ShouldIgnoreTouchEmulatedMessage() only works for legacy mouse events, see https://docs.microsoft.com/en-us/windows/win32/tablet/system-events-and-mouse-messages